### ISSUE TITLE
Discussion: implement the platform-specific clients as a wrapper around surf?

### ISSUE THREAD

**Issue by tomhoule** [graphql-rust/graphql-client#272]
It would need to be a wrapper - request and response configuration do not vary, and we want to link variables, queries and response shapes.

We would get:

- an independently maintained http client stack
- cross-platform support (web and native for now) out of the box

Cons:

- As far as I am aware of, surf is async-first. We might still want to provide a sync client for convenience

**Comment by h-michael** [comment]
@tomhoule Is the platform-specific client you are referring to now referring to the graphql_client_web?

**Comment by tomhoule** [comment]
So I think we could replace the web client we have now with something more robust and maintained by other people, and at the same time this would give us a native HTTP client. That way we only need to interface with a single library (surf).

**Comment by h-michael** [comment]
> and at the same time this would give us a native HTTP client.

At present, users can send requests using their favorite client library, but does that mean that we will provide a feature build that includes clients?

**Comment by tomhoule** [comment]
Yes it should definitely be a feature (on by default or opt-in, I'm not sure). Users should definitely have access to the current API too in my opinion.

**Comment by h-michael** [comment]
I think that it is a good configuration that there is room for the current user to select the client library by himself.
If you do not add a new function with the client, I think that the built-in client can be opt-in.
